build-cluster-ec2.yml

```diff
@@ -106,14 +207,37 @@
   tasks:
     - include: python/tasks/main.yml
     - include: java/tasks/main.yml
+    - include: ntp/tasks/main.yml
     - include: docker/tasks/main.yml
       when: "{{ INSTALL_DOCKER }} == true"
+    - include: conductr/tasks/install-agent.yml
+    - include: bundle/tasks/config-elasticsearch.yml
+
+- name: Setup Public Nodes
+  hosts: proxy_nodes
+  user: "{{ REMOTE_USER }}"
+  gather_facts: False
+  sudo: True
+
+  vars:
+# These are the default ConductR Agent roles - these roles may be overridden within the VARS_FILE
+    CONDUCTR_AGENT_ROLES:
+      - haproxy
+
+  vars_files:
+    - "{{ VARS_FILE }}"
+
+  tasks:
+    - include: python/tasks/main.yml
+    - include: java/tasks/main.yml
     - include: ntp/tasks/main.yml
+    - include: docker/tasks/main.yml
+      when: "{{ INSTALL_DOCKER }} == true"
     - include: haproxy/tasks/main.yml
-    - include: conductr/tasks/install-core.yml
     - include: conductr/tasks/install-agent.yml
```

Comment:
If we have private nodes declared as `CONDUCTR_PUBLIC_AGENT_ROLES`, we can then pass the roles into the agent task as such:

```
- include: conductr/tasks/install-agent.yml
   vars:
     CONDUCTR_AGENT_ROLES={{ CONDUCTR_PUBLIC_AGENT_ROLES }}
```